staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -197,7 +212,9 @@ func (f *featureGate) Enabled(key Feature) bool {
 
 // AddFlag adds a flag for setting global feature gates to the specified FlagSet.
 func (f *featureGate) AddFlag(fs *pflag.FlagSet) {
+	f.lock.Lock()
 	f.closed = true
+	f.lock.Unlock()
```

Comment:
I strongly prefer the `defer` to unlock.  Some reason I can't?  I'm betting something else is trying to grab it underneath.  I really hate the golang sync library.

Reply:
both KnownFeatures and the flag set up make use of methods that RLock